predator/asp/greedy-target-seed-solving_SA.lp

```diff
@@ -46,32 +39,23 @@ present(M) :- any_seed(M).
 % % a metabolite is present or not
 % { present(M) } :- metabolite(M).
 
-% % products of a feasible reaction are producible
-% producible(M) :- feasible(R), product(M,R).
+% products of a feasible reaction are producible
+producible(M) :- feasible(R), product(M,R).
 
 % % every present metabolite must be producible
 % :- present(M), not producible(M), not any_seed(M).
 
 %%%%%%% version 2 %%%%%%%%
 
-% a metabolite is present if it is given as a seed or if it is the product of a feasible reaction
-present(X) :- product(X,R), feasible(R).
```

Comment:
Not sure if this helps? The atoms of predicate `present` are chosen and `feasible` depends on which metabolites are present. I think it does not hurt either but in doubt I prefer less code.